records/state/state_test.go

```diff
@@ -39,62 +41,146 @@ func TestPID_UnmarshalJSON(t *testing.T) {
 	}
 }
 
-func TestTask_containerIP(t *testing.T) {
-	makeTask := func(networkInfoAddress string, label Label) Task {
-		labels := make([]Label, 0, 1)
-		if label.Key != "" {
-			labels = append(labels, label)
+func TestTask_IPs(t *testing.T) {
+	for i, tt := range []struct {
+		*Task
+		srcs []string
+		want []net.IP
+	}{
+		{nil, nil, nil},
+		{nil, []string{}, nil},
+		{nil, []string{"host"}, nil},
+		{ // no IPs for the given sources
+			Task: task(statuses(status(state("TASK_RUNNING"), netinfo("1.2.3.4")))),
+			srcs: []string{"host", "mesos"},
+			want: nil,
+		},
+		{ // unknown IP sources are ignored
+			Task: task(statuses(status(state("TASK_RUNNING"), netinfo("1.2.3.4")))),
+			srcs: []string{"foo", "netinfo", "bar"},
+			want: ips("1.2.3.4"),
+		},
+		{ // source order
+			Task: task(
+				slaveIP("2.3.4.5"),
+				statuses(status(state("TASK_RUNNING"), netinfo("1.2.3.4"))),
+			),
+			srcs: []string{"host", "netinfo"},
+			want: ips("2.3.4.5", "1.2.3.4"),
+		},
+		{ // statuses state
+			Task: task(
+				statuses(
+					status(state("TASK_RUNNING"), netinfo("1.2.3.4")),
+					status(state("TASK_STOPPED"), netinfo("2.3.4.5")),
+				),
+			),
+			srcs: []string{"netinfo"},
+			want: ips("1.2.3.4"),
+		},
+		{ // statuses ordering
+			Task: task(
+				statuses(
+					status(state("TASK_RUNNING"), netinfo("1.2.3.4")),
```

Comment:
I think I'm OK with aggregating across source types. Across statuses though (from newest to oldest) seems super-fragile, mostly because we have no way of knowing which newer IPs are replacements for others, and is new behavior that does not belong in this PR.